### PR TITLE
RBAC Content Guards now use roles

### PR DESCRIPTION
**Pull request by gerrod3**
This PR requires https://github.com/pulp/pulpcore/pull/1627/. I don't know if I should include a migration to change all the current permission holders over to their new roles equivalent.

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/app/serializers/publication.py [hunk 1]
```diff
@@ -86,6 +86,7 @@ class RBACContentGuardSerializer(ContentGuardSerializer):
     @extend_schema_field(GroupUserSerializer(many=True))
     def get_users(self, obj):
         """Finds all the users with this object's download permission."""
+        # TODO re-implement this util function with new Role framework
```
Comment (mdellweg) [inline pulpcore/app/serializers/publication.py:89]: @gerrod3 can you verify, this is what it needs?

https://github.com/pulp/pulpcore/pull/1627/commits/f7b1b2ea191672f734387e87cceb4470921f237e

Comment (gerrod3) [reply pulpcore/app/serializers/publication.py:89]: Yes, they look good!

**Review by mdellweg — APPROVED** [review]

File: pulpcore/tests/functional/api/using_plugin/utils.py [hunk 2]
```diff
@@ -217,3 +219,43 @@ def del_user(user, cfg=config.get_config()):
         cli_client,
         "\n".join(DELETE_USER_CMD).format(**user),
     )
+
+
+def gen_user_rest(cfg=config.get_config(), model_roles=None, object_roles=None, **kwargs):
```
Comment (mdellweg) [inline pulpcore/tests/functional/api/using_plugin/utils.py:224]: I think calling functions for defaults is an anti-pattern.

File: pulpcore/tests/functional/api/using_plugin/utils.py [hunk 3]
```diff
@@ -217,3 +219,43 @@ def del_user(user, cfg=config.get_config()):
...
+    user = admin_user_api.create(user_body)
+
+    if model_roles:
+        for role in model_roles:
+            user_role = {"role": role, "content_object": None}
+            admin_user_roles_api.create(user.pulp_href, user_role)
+    if object_roles:
+        for role, obj in object_roles:
+            user_role = {"role": role, "content_object": obj}
+            admin_user_roles_api.create(user.pulp_href, user_role)
```
Comment (mdellweg) [inline pulpcore/tests/functional/api/using_plugin/utils.py:246]: :+1:

File: pulpcore/tests/functional/api/using_plugin/utils.py [hunk 4]
```diff
@@ -217,3 +219,43 @@ def del_user(user, cfg=config.get_config()):
...
+
+    # Set password on UserResponse for utility
+    user.openapi_types["password"] = "str"
+    user.attribute_map["password"] = "password"
+    setattr(user, "password", user_body["password"])
```
Comment (mdellweg) [inline pulpcore/tests/functional/api/using_plugin/utils.py:251]: Let me see if i understand that. The openapi does not return the password (write-only field), and so you are faking that it did?

Comment (mdellweg) [reply pulpcore/tests/functional/api/using_plugin/utils.py:251]: What if we simply returned the dict that the tests expect anyway here?

File: pulpcore/tests/functional/api/using_plugin/test_content_guard.py [hunk 5]
```diff
@@ -84,26 +94,32 @@ def test_03_only_creator_access(self):
...
 
     def test_05_remove_users(self):
         """Use the /remove/ endpoint to remove users permission to access distribution"""
-        body = {"usernames": (self.user_a["username"], self.user_b["username"])}
-        self.creator_user["rbac_guard_api"].remove_permission(self.distro.content_guard, body)
+        body = {
+            "users": (self.user_a["username"], self.user_b["username"]),
+            "role": self.DOWNLOAD_ROLE,
+        }
+        self.creator_user["rbac_guard_api"].remove_role(self.distro.content_guard, body)
```
Comment (mdellweg) [inline pulpcore/tests/functional/api/using_plugin/test_content_guard.py:110]: I must say, i am kind of pleased by this interface.

**Review by mdellweg — APPROVED** [review]

File: CHANGES/9498.feature [hunk 6]
```diff
@@ -0,0 +1 @@
+Reworked RBAC Content Guards to now use roles.
```
Comment (bmbouter) [inline CHANGES/9498.feature:1]: Can there be a callout of the APIs that this adds, even just the top-level endpoint?

Also what about adding a 9498.removal outlining the API calls that are removed and that they were tech preview.

Comment (mdellweg) [reply CHANGES/9498.feature:1]: I think, we only removed stuffs.

Comment (mdellweg) [reply CHANGES/9498.feature:1]: right. And a new permission to serve it.

**Review by bmbouter — APPROVED** [review]
Thanks!

**Pull request merged**